docs/proposals/multi-arch.md

```diff
@@ -0,0 +1,114 @@
+# Multi-arch image support for ORAS
+
+## Command Design
+
+### Overview
+
+- `oras manifest index create`: Create an image index from source manifests. 
+- `oras manifest index update`: Add/remove manifests to/from an image index.
+
+### Create an image index
+
+#### Definition
+
+Create an image index from source manifests. The command auto-detects platform information for each source manifests.
+
+> [!IMPORTANT]
+> All source manifests referenced are required to exist in the same repository as the target index.
+
+Usage:
+```
+oras manifest index create [flags] <name>[:<tag[,<tag>][...]] [{<tag>|<digest>}...]
```

Comment:
Asking because `buildx` generates index even for single platform images build, which includes both attestation and image manifests. The attestation contains annotations like
`
      "annotations": {
        "vnd.docker.reference.digest": "sha256:xxx",
        "vnd.docker.reference.type": "attestation-manifest"
      }
`

If we don't flatten index, then such annotation will be lost during the index creation.

Reply:
Had a discussion with @shizhMSFT offline. Index manifest input should not be handled as a special case. ORAS is a general registry client and there is no deterministic way to check if the user intention. Can add `oras manifest index merge` if needed.